### DISCUSSION TITLE
Issues setting up dev enviornment

### DISCUSSION THREAD

**Discussion by annaleighsmith** [Eventual-Inc/Daft#2230] [Q&A]
When I run `make .venv` I keep getting the following errors:

```
.venv/bin/uv pip install -r requirements-dev.txt
  × No solution found when resolving dependencies:
  ╰─▶ Because ray[client]==2.10.0 is unusable because no wheels are available with a matching Python ABI and you require ray[client]==2.10.0, we can conclude that the requirements are
      unsatisfiable.
```

I have Python 3.12.3 and I am running this on Arch linux. Has anyone had similar problems?

**Comment by jaychia** [comment]
Hi @annaleighsmith !

I looked at Ray’s 2.10 release and it seems they don’t have a wheel for Python 3.12: https://pypi.org/project/ray/2.10.0/#files

You could either try downgrading your version of Python, or we can maybe bump the version of Ray that we are using for development in requirements-dev.txt to 2.12, which seems to have Python 3.12 support (https://pypi.org/project/ray/2.12.0/#files) :)
